Tutorial/jsonscheme (#114)
gardener/landscaper: docs/usage/Blueprints.md

The import definitions describe the data that needed by the blueprint.<br>
The export definitions describe the data that is produced by the blueprint.<br>

Imports and Exports always consists of a name (_MUST_ be unique for the blueprint) and can be generally described as _Target_-Im/Exports or _Data_-Im/Exports.

```yaml
imports:
- name: <string> # some unique name
  required: <boolean> # defaults to true
```

:warning: in the following data import and target import is used as synonym for import and export. The specification applies to both.

#### Data Imports

Data imports are the default import type that can describe arbitrary data in json or yaml format.
[JSONSchema](https://json-schema.org/) is used to describe the structure fo the data.

It is recommended to provide a description and an example for the structure, so that users of the blueprint know what to provide (see the [json docs](http://json-schema.org/understanding-json-schema/reference/generic.html#annotations)).

```yaml
imports:
- name: my-import
  schema:
    <jsonschema>
```

For detailed information about the jsonschema and landscaper specifics see [JSONSchema Docs](./JSONSchema.md)

#### Targets

> Target are a specific type of import that contains additional meaning for the deployers.

Reviewer: ```suggestion
Targets are a specific type of import and will information will be interpreted by deployers.
```

> This means that targets should define the target environment where a deployer installs/deploys software to.

Reviewer: ```suggestion
The concept of a Target is to define the environment where a deployer installs/deploys software.
```
